consensus/src/processes/traversal_manager.rs

```diff
@@ -31,20 +31,32 @@ impl<T: GhostdagStoreReader, U: ReachabilityStoreReader, V: RelationsStoreReader
         Self { genesis_hash, ghostdag_store, relations_store, reachability_service }
     }
 
-    pub fn calculate_chain_path(&self, from: Hash, to: Hash) -> ChainPath {
+    pub fn calculate_chain_path(&self, from: Hash, to: Hash, max_traversal_allowed: usize) -> ChainPath {
         let mut removed = Vec::new();
         let mut common_ancestor = from;
-        for current in self.reachability_service.default_backward_chain_iterator(from) {
+        for current in self.reachability_service.default_backward_chain_iterator(from).take(max_traversal_allowed) {
```

Comment:
This iteration would start from block at `from + max_traversal_allowed` and go back all the way to `from` correct?

Reply:
well technically it is more like starting at `from`, and going back at most `max_traversal_allows` (the break condition is in the loop).